File: 1698/1698.c
int numberOfDistinctSubstringsInAS(struct TreeNode* root) {
    if (!root) return 0;
    int l = numberOfDistinctSubstringsInAS(root->left);
    int r = numberOfDistinctSubstringsInAS(root->right);
    return 1 + (l > r ? l : r);
}

File: 1698/1698.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var numberOfDistinctSubstringsInAS = function(root) {
    if (!root) return 0;
    return 1 + Math.max(numberOfDistinctSubstringsInAS(root.left), numberOfDistinctSubstringsInAS(root.right));
};